mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/task/SolveMessageInconsistenciesService.java

```diff
@@ -19,25 +19,362 @@
+            return messageIdDAO.updateMetadata(messageFromImapUid)
+                .doOnSuccess(any -> notifySuccess(context))
+                .thenReturn(Task.Result.COMPLETED)
+                .onErrorResume(error -> {
+                    notifyFailure(context);
+                    return Mono.just(Task.Result.PARTIAL);
+                });
+        }
+
+        private void notifyFailure(Context context) {
+            context.incrementErrors();
+            LOGGER.error("Error when fixing inconsistency for outdated message: {}", messageFromMessageId.getComposedMessageId());
+        }
+
+        private void notifySuccess(Context context) {
+            LOGGER.info("Inconsistency fixed for outdated message: {}", messageFromMessageId.getComposedMessageId());
+            context.incrementUpdatedMessageIdEntries();
+            context.addFixedInconsistency(messageFromMessageId.getComposedMessageId());
+        }
+    }
+
+    private static class OrphanMessageIdEntry implements Inconsistency {
+        private final ComposedMessageIdWithMetaData message;
+
+        private OrphanMessageIdEntry(ComposedMessageIdWithMetaData message) {
+            this.message = message;
+        }
+
+        @Override
+        public Mono<Task.Result> fix(Context context, CassandraMessageIdToImapUidDAO imapUidDAO, CassandraMessageIdDAO messageIdDAO) {
+            return messageIdDAO.delete((CassandraId) message.getComposedMessageId().getMailboxId(), message.getComposedMessageId().getUid())
+                    .doOnSuccess(any -> notifySuccess(context))
+                    .then(Mono.just(Task.Result.COMPLETED))
+                .single()
+                .onErrorResume(error -> {
+                    notifyFailure(context);
+                    return Mono.just(Task.Result.PARTIAL);
+                });
+        }
+
+        private void notifyFailure(Context context) {
+            context.incrementErrors();
+            LOGGER.error("Error when fixing inconsistency for orphan message in MessageId: {}", message.getComposedMessageId());
+        }
+
+        private void notifySuccess(Context context) {
+            LOGGER.info("Inconsistency fixed for orphan message in MessageId: {}", message.getComposedMessageId());
+            context.incrementRemovedMessageIdEntries();
+            context.addFixedInconsistency(message.getComposedMessageId());
+        }
+    }
+
+    static class Context {
+        static class Builder {
+            private Optional<Long> processedImapUidEntries;
+            private Optional<Long> processedMessageIdEntries;
+            private Optional<Long> addedMessageIdEntries;
+            private Optional<Long> updatedMessageIdEntries;
+            private Optional<Long> removedMessageIdEntries;
+            private ImmutableList.Builder<ComposedMessageId> fixedInconsistencies;
+            private Optional<Long> errors;
```

Comment:
Can we rather have `private ImmutableList.Builder<ComposedMessageId> errors;` so that an admin can easily know what failed?